src/test/java/edu/iis/mto/bsearch/BinarySearchTest.java

```diff
@@ -1,52 +1,96 @@
+        SearchResult result;
+        try {
+            result = BinarySearch.search(sequenceShort[0], sequenceShort);
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+            assertEquals(e.getMessage(), "");
+            return;
+        }
         assertTrue(result.isFound());
         assertEquals(0, result.getPosition());
     }
 
     @Test
-    void notInShortSequence() {
-        assertFalse(BinarySearch.search(sequenceShort[0]+1, sequenceShort).isFound());
+    public void notInShortSequence() {
+        try {
+            assertFalse(BinarySearch.search(sequenceShort[0]+1, sequenceShort).isFound());
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+            assertEquals(e.getMessage(), "");
+        }
     }
 
     @Test
-    void firstInSequence() {
-        SearchResult result = BinarySearch.search(sequenceLong[0], sequenceLong);
+    public void firstInSequence() {
+        SearchResult result;
+        try {
+            result = BinarySearch.search(sequenceLong[0], sequenceLong);
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+            assertEquals(e.getMessage(), "");
+            return;
+        }
         assertTrue(result.isFound());
         assertEquals(0, result.getPosition());
     }
 
     @Test
-    void lastInSequence() {
-        SearchResult result = BinarySearch.search(sequenceLong[sequenceLong.length-1], sequenceLong);
+    public void lastInSequence() {
+        SearchResult result;
+        try {
+            result = BinarySearch.search(sequenceLong[sequenceLong.length-1], sequenceLong);
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+            assertEquals(e.getMessage(), "");
+            return;
+        }
         assertTrue(result.isFound());
         assertEquals(sequenceLong.length-1,  result.getPosition());
     }
 
     @Test
-    void midInSequence() {
-        SearchResult result = BinarySearch.search(sequenceLong[sequenceLong.length/2], sequenceLong);
+    public void midInSequence() {
+        SearchResult result;
+        try {
+            result = BinarySearch.search(sequenceLong[sequenceLong.length/2], sequenceLong);
+        } catch (IllegalAccessException e) {
+            assertEquals(e.getMessage(), "");
+            e.printStackTrace();
+            return;
+        }
         assertTrue(result.isFound());
         assertEquals(sequenceLong.length/2, result.getPosition());
     }
 
     @Test
-    void notInLongSequence() {
-        assertFalse(BinarySearch.search(sequenceLong[0]-1, sequenceLong).isFound());
+    public void notInLongSequence() {
+        try {
+            assertFalse(BinarySearch.search(sequenceLong[0]-1, sequenceLong).isFound());
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+            assertEquals(e.getMessage(), "");
+        }
+    }
+
+    @Test
+    public void emptySequence() {
```

Comment:
jeden test weryfikujący czy faktycznie wyjątek jest rzucany przy określonych warunkach. Pozostałe testy w ogóle nie powinny mieć try catch - to zmiejsza czytelność testów.